subsys/bluetooth/host/hci_core.h

```diff
@@ -109,11 +137,17 @@ struct bt_dev {
 	bt_addr_le_t		id_addr[CONFIG_BT_ID_MAX];
 	u8_t                    id_count;
 
-	/* ID Address used for advertising */
-	u8_t                    adv_id;
+	struct bt_conn_le_create_param create_param;
 
+#if !defined(CONFIG_BT_ADV_EXT)
+	/* Legacy advertiser */
+	struct bt_le_ext_adv    adv;
+#else
+	/* Pointer to reserved advertising set */
+	struct bt_le_ext_adv    *adv;
+#endif
```

Comment:
Wouldn't it make sense to keep this a pointer in both cases, since that avoids #ifdefs in the code. When `CONFIG_BT_ADV_EXT` is disabled you'd initialise this to point to a singleton (static variable in `hci_core.c`)

Reply:
It will add code size for the simple, usual case though won't it? I am worried we keep piling up small code size increases. That said, this is internal, can be discussed after rc1.

Reply:
What if the single-instance was changed to `struct bt_le_ext_adv adv[1]`? That way you can treat it as a pointer.

Reply:
> What if the single-instance was changed to `struct bt_le_ext_adv adv[1]`? That way you can treat it as a pointer.

Maybe I misunderstand, but I would still need an ifdef here then. But `bt_adv_lookup_legacy` could be simplified, or removed. Is that what you mean?